### PR TITLE
Add robots.txt support

### PR DESCRIPTION
**Pull request by nein09**
For https://github.com/gleanerio/gleaner/issues/27

Main features:
- Adds support for using a robots.txt file as a source url in the config, and a new `sourcetype`, called `robots`. This will get a sitemap, or list of sitemaps, from the robots.txt file that is given in the source config url, like so:
```
- name: npdc
  sourcetype: robots
  headless: false
  url: https://npdc.nl/robots.txt
  properName: Netherlands Polar Data Center
  domain: https://npdc.nl
  active: true
```
- Looks for a robots.txt file on a domain _even if one was not given in the _config_. If it finds one, it modifies crawl behavior to respect the crawl delay and allowed / disallowed paths in the robots.txt file

Other stuff that ended up in here:
- Adds a utility function to fetch and parse a robots.txt, given its url
- Adds some helper functions to the sources config class
- Removes a mystery `sleep`
- Addresses a todo: makes `resources.ResourceUrl` return an `err`

### CODE REVIEW

File: internal/config/sources.go [hunk 1]
```diff
@@ -162,6 +163,26 @@ func GetActiveSourceByType(sources []Sources, key string) []Sources {
 	return sourcesSlice
 }
 
+func GetActiveSourceByHeadless(sources []Sources, headless bool) []Sources {
```
Comment (valentinedwv) [inline internal/config/sources.go:166]: ;)

**Review by valentinedwv — APPROVED** [review]
Looks good.
Tests... oh look tests.. guess I need to add that to the automation

File: internal/config/sources.go (hunk 1 above)
Comment (nein09) [reply internal/config/sources.go:166]: It just seemed easier than re-checking them every time, heh.

**Review by fils — APPROVED** [review]
Looks nice to me.  Thanks for the test and some better return values and error handling in there too!

**Comment by nein09** [comment]
@fils @valentinedwv I think there's a huge bug in here somewhere, because I just tried indexing on the dev branch, and things did NOT go as expected; nothing was crawled at all. I'm looking into what happened.

**Comment by nein09** [comment]
Of course, the first thing I tried was to revert https://github.com/gleanerio/gleaner/pull/71/commits/82ccb78486379e7441697b913c4ad4b7d6d38666 and of course things are going smoothly now. Lesson of the day: let sleeping sleeps lie?

**Comment by nein09** [comment]
I'm going to do a bit more testing to confirm that this really is what happened, and open another PR if it is.

**Pull request merged**